app/models/solidus_stripe/payment_intent.rb

```diff
@@ -23,6 +23,15 @@ def self.prepare_for_payment(payment)
       intent
     end
 
+    def self.retrieve_last_usable_intent(payment)
+      intent = where(payment_method: payment.payment_method, order: payment.order).last
+      intent if intent&.usable?
+    end
+
+    def usable?
+      stripe_intent_id && stripe_intent.status == 'requires_payment_method'
```

Comment:
Thoughts on checking the amount too?

Reply:
Yes! You're right. This could generate dangerous problems. Thanks @elia I'll verify the amount ⏳